File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/interfaces/behaviour/view/BehaviourMainView.java
package com.nerdscorner.mvplib.testapp.interfaces.behaviour.view;

import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.nerdscorner.mvplib.interfaces.view.BaseActivityView;
import com.nerdscorner.mvplib.testapp.R;
import com.nerdscorner.mvplib.testapp.interfaces.behaviour.presenter.BehaviourMainPresenterInterface;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class BehaviourMainView extends BaseActivityView<BehaviourMainPresenterInterface> implements BehaviourMainViewInterface {
    @BindView(R.id.text) TextView textView;

    public BehaviourMainView(AppCompatActivity activity) {
        super(activity);
        ButterKnife.bind(this, activity);
    }

    @OnClick(R.id.some_button)
    public void onActionClicked() {
        getPresenter().onActionClicked();
    }

    @Override
    public void setTextValue(CharSequence value) {
        textView.setText(value);
    }
}

File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/behaviour/view/BehaviourMainView.kt
package com.nerdscorner.mvplib.testapp.events.behaviour.view

import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.nerdscorner.mvplib.events.view.BaseActivityView
import com.nerdscorner.mvplib.testapp.R

class BehaviourMainView(activity: AppCompatActivity) : BaseActivityView(activity) {
    var textView: TextView = activity.findViewById(R.id.text)

    init {
        onClick(R.id.some_button, ActionClickedEvent())
    }

    fun setTextValue(value: CharSequence) {
        textView.text = value
    }

    class ActionClickedEvent
}
